### PR TITLE
[FEATURE] Enable self-intializing capabilities for `ExpectColumnValuesToMatchRegex`/`ExpectColumnValuesToNotMatchRegex`

### PR DESCRIPTION
**Pull request by cdkini**
Changes proposed in this pull request:
- Enable self-initializing capabilities for regex-based expectations
- Integration tests for both

### Definition of Done
Please delete options that are not relevant.

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have run any local integration tests and made sure that nothing is broken.

### CODE REVIEW

File: tests/integration/profiling/rule_based_profilers/test_profiler_user_workflows.py [hunk 1]
```diff
@@ -293,6 +342,60 @@ def test_alice_profiler_user_workflow_single_batch(
...
+    expectation_config_kwargs: dict = result.expectation_config.kwargs
+    assert expectation_config_kwargs == {
+        "auto": True,
+        "batch_id": "cf28d8229c247275c8cc0f41b4ceb62d",
+        "column": "id",
+        "include_config": True,
+        "mostly": 1.0,
+        "regex": "(?:[A-Fa-f0-9]){0,4}(?: ?:? ?(?:[A-Fa-f0-9]){0,4}){0,7}",
+        "result_format": "SUMMARY",
+    }
```
Comment (NathanFarmer) [inline tests/integration/profiling/rule_based_profilers/test_profiler_user_workflows.py]: Missing type hints

File: tests/integration/profiling/rule_based_profilers/test_profiler_user_workflows.py [hunk 2]
```diff
@@ -293,6 +342,60 @@ def test_alice_profiler_user_workflow_single_batch(
...
```
Comment (NathanFarmer) [inline tests/integration/profiling/rule_based_profilers/test_profiler_user_workflows.py]: Missing type hints

**Review by NathanFarmer — CHANGES_REQUESTED** [review]
Looking good, just some type hints!

File: tests/rule_based_profiler/parameter_builder/test_regex_pattern_string_parameter_builder.py [hunk 3]
```diff
@@ -68,16 +68,16 @@ def test_regex_pattern_string_parameter_builder_instantiation_with_defaults(
     data_context: DataContext = mock_data_context
 
     candidate_regexes: Set[str] = {
-        r"/\d+/",  # whole number with 1 or more digits
```
Comment (Shinnnyshinshin) [inline tests/rule_based_profiler/parameter_builder/test_regex_pattern_string_parameter_builder.py:71]: 🙇🏼

**Review by alexsherstinsky — APPROVED** [review]
@cdkini Good to go, once the feedback from @NathanFarmer is incorporated (and any other relevant change requests).  Thanks!!!

File: tests/integration/profiling/rule_based_profilers/test_profiler_user_workflows.py (hunk 1 above)
Comment (cdkini) [reply tests/integration/profiling/rule_based_profilers/test_profiler_user_workflows.py]: done! sorry I was being lazy here I appreciate the diligence around type annotating 😅

File: tests/integration/profiling/rule_based_profilers/test_profiler_user_workflows.py (hunk 2 above)
Comment (cdkini) [reply tests/integration/profiling/rule_based_profilers/test_profiler_user_workflows.py]: done!

File: tests/rule_based_profiler/parameter_builder/test_regex_pattern_string_parameter_builder.py (hunk 3 above)
Comment (cdkini) [reply tests/rule_based_profiler/parameter_builder/test_regex_pattern_string_parameter_builder.py:71]: oh regex why are you so confusing

**Review by NathanFarmer — APPROVED** [review]
LGTM!

**Pull request merged**